common/lib/common-definitions/src/logger.ts

```diff
@@ -40,7 +54,7 @@ export interface ITelemetryProperties {
  * @param category - category of the event, like "error", "performance", "generic", etc.
  * @param eventName - name of the event.
  */
-export interface ITelemetryBaseEvent extends ITelemetryProperties {
+export interface ITelemetryBaseEvent extends ITelemetryBaseProperties {
```

Comment:
I'd add some note in breaking.md that implementers of ITelemetryBaseLogger may need to use ITelemetryBaseProperties instead of ITelemetryProperties (same for any types that are comprising ITelemetryBaseEvent, i.e. TelemetryBaseEventPropertyType, etc.
Otherwise, people may struggle to understand what changed and how they should react (i.e. consumers may think they need to support arrays now)